src/access/universal.test.ts

```diff
@@ -1268,3 +1271,113 @@ describe("getGroupAccessAll", () => {
     expect(access).toBeNull();
   });
 });
+
+describe("getAccessFor", () => {
+  it("calls to getAgentAccess with the appropriate parameters", async () => {
+    const universalModule = jest.requireActual("./universal") as {
+      getAgentAccess: () => Promise<Access | null>;
+    };
+    universalModule.getAgentAccess = jest.fn();
```

Comment:
Hmm, so I take it `spyOn` still doesn't work here? Either way, if this approach does work, do we still need the separate module?

One other question: how does this overwriting of a property on `universalModule` affect other tests? Specifically, if you were to add another test that also calls code that calls out to `getAgentAccess`, will that hit this mock function or the actual one? If the former, I think that's something we'll want to prevent now, rather than hitting something really hard to debug later. (Which I suppose we could do by storing the old value and restoring it at the end of the `it`.)

Reply:
I don't think this impacts other tests, because what gets mocked here is the exported function, and not its reference that gets used internally in the module (which is the reason why this doesn't work if `getAccessFor` is in the same module as `getAgentAccess`. And regardless, in an upcoming PR (https://github.com/inrupt/solid-client-js/commit/0f8b532f6ba867a0ebd5b84d8fb8e5b830985a01) I already moved the tests mocking the `universal` module in a different file, because it no longer made sense to test `getAccessFor` in the same test file as `getAgentAccess` and the like, so this won't be a problem anyway very soon.

Reply:
Ah perfect! And then maybe even the `spyOn`s will work again? :)